### PR TITLE
Refactor pathfinding element to a subsystem, should fix some pathfinding bugs

### PR DESCRIPTION
**Pull request by BraveMole**
## About The Pull Request

Make pathfinding a ss only, no need for that element shitcode. Remove all these assoc list calls.

closes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/8727
closes: https://github.com/tgstation/TerraGov-Marine-Corps/issues/8788 (probably, not sure about this one)

Also remove the unused internal code of many bots, as well as the old tg Astar

## Why It's Good For The Game

Possibly better perf, and that element was really useless

## Changelog
:cl:
fix: Minions no longer move when unconcious or dead
fix: Minions no longer stuck by opened req railing
refactor: Refactor pathfinding
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/pathfinder.dm [hunk 1]
```diff
@@ -1,54 +1,97 @@
...
-	run = 0
-	free = 1
-	flow = new /list(lcount)
+///Signal handler to change the target's atom_to_walk_to
+/datum/controller/subsystem/pathfinder/proc/change_target(mob/target, atom/atom_to_walk_to)
+	SIGNAL_HANDLER
+	for(var/datum/pathfinding_datum/pathfinding_datum AS in pathfinding_datums_list)
+		if(pathfinding_datum.mob_parent == target)
+			pathfinding_datum.atom_to_walk_to = atom_to_walk_to
+			return
```
Comment (TiviPlus) [inline code/controllers/subsystem/pathfinder.dm:74]: this stinks

File: code/controllers/subsystem/pathfinder.dm [hunk 2]
```diff
@@ -1,54 +1,97 @@
...
 
+///Remove a target mob from the pathfinding system
+/datum/controller/subsystem/pathfinder/proc/remove_from_pathfinding(mob/target)
+	UnregisterSignal(target, COMSIG_PATHFINDER_SET_ATOM_TO_WALK_TO)
+	for(var/datum/pathfinding_datum/pathfinding_datum AS in pathfinding_datums_list)
+		if(pathfinding_datum.mob_parent == target) //We don't break here, to ensure there is no duplicate being left
+			pathfinding_datums_list -= pathfinding_datum
```
Comment (TiviPlus) [inline code/controllers/subsystem/pathfinder.dm:81]: this too

**Comment by TiviPlus** [comment]
Also why remove old bot code completely ://

File: code/controllers/subsystem/pathfinder.dm (hunk 1 above)
Comment (BraveMole) [reply code/controllers/subsystem/pathfinder.dm:74]: no, same perf as an assoc list

**Pull request merged**